### PR TITLE
gnome: Fix minimum gresource dependency required

### PR DESCRIPTION
**Pull request by nirbheek**
This is the latest release of glib that exists and has the required dependency-generation fixes.

Without this GNOME Recipes cannot even configure.

### CODE REVIEW

File: mesonbuild/modules/gnome.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 #
 # https://github.com/ninja-build/ninja/issues/1184
 # https://bugzilla.gnome.org/show_bug.cgi?id=774368
-gresource_dep_needed_version = '>= 2.52.0'
+gresource_dep_needed_version = '>= 2.51.2'
```
Comment (ignatenkobrain) [inline mesonbuild/modules/gnome.py]: why?

Comment (nirbheek) [inline mesonbuild/modules/gnome.py]: As the commit message says, the version here is wrong:

```
This is the latest release of glib that exists and has
the required dependency-generation fixes.
```

Also from IRC:
```
<nirbheek> TingPing, 2.51.2 has the fix right?
<TingPing> yes
```

Comment (TingPing) [inline mesonbuild/modules/gnome.py]: Turns out 2.51.1 has it also. Though honestly I don't think this matters *that* much, its not like anybody is releasing software depending on a dev release.

Comment (nirbheek) [inline mesonbuild/modules/gnome.py]: I've updated the version to `2.51.1`. Doesn't cost anything to be accurate...

**Pull request merged**